lib/web.ts

```diff
@@ -230,7 +215,41 @@ export default class AutoConsent {
       });
     }
 
-    return allFoundCmps;
+    return {
+      normal,
+      cosmetic,
+    };
+  }
+
+  async detectPopups(cmps: AutoCMP[]): Promise<AutoCMP[]> {
+    const popupLookups: Promise<AutoCMP>[] = [];
+    for (const cmp of cmps) {
+      popupLookups.push(this.waitForPopup(cmp).then((isOpen) => {
+        if (isOpen) {
+          this.updateState({ detectedPopups: this.state.detectedPopups.concat([cmp.name]) });
+          this.sendContentMessage({
+            type: 'popupFound',
+            cmp: cmp.name,
+            url: location.href,
+          }); // notify the browser
+          return cmp;
+        } else {
+          return Promise.reject(`${cmp.name} popup not found`);
+        }
+      }));
+    }
+
+    const result: AutoCMP[] = [];
+    // could use `Promise.allFulfilled`, but it is often unavailable in polyfilled environments
+    for (const popupLookup of popupLookups) {
+      try {
+        result.push(await popupLookup);
```

Comment:
Is this potentially an unhandled rejection? If the 2nd cmp in the list is not found, and the check runs faster than the 1st we won't `await` the promise until after it has rejected.

Reply:
Yeah, I think you're right, we may produce some console errors in this case, although it should still work.
This promise handling feels overcomplicated, I'll see if I can simplify it a bit

Reply:
After taking another look, I'm pretty sure I forgot to remove the `break;` because `detectPopups()` is supposed to return an array. Thanks for catching!

Reply:
I think there is still an issue that we're not handling all the promises in `popupLookups` in the first tick, as you're `await`ing in a loop. Perhaps we can just combine everything into the initial promise:

```javascript
const result = [];
const popupLookups = cmps.map(cmp => this.waitForPopup(cmp).then((isOpen) => {
  if (isOpen) result.push(cmp)
}).catch(() => {}); 
// As we've silence errors, we can safely use `Promise.all`
await Promise.all(popupLookup);
return result;
```

Alternatively, if we want to be as fast as possible, we could turn `detectPopups` into an async generator, and use `Promise.race` to emit the first open popup ASAP.

Reply:
I'm a bit cautious about `Promise.race()` because we already had problems with `Promise.any()` not being implemented in polyfilled environments